content/articles/creating-a-pyspark-data-frame-a-beginners-guide/index.md

```diff
@@ -0,0 +1,159 @@
+2. Have a python IDE or follow this [documentation](https://www.python.org/) to install.
+### Dataframe creation
+It can be created in one of the following two ways:
+
+1. From a Resilient Distributed Dataset.
+2. Getting information from outside sources, like CSV.
+
+Installing the PySpark library will be our first move.
+```Python
+!pip install pyspark
+```
+### Using an existing Resilient Distributed Dataset to make a PySpark data frame
+`Parallelize()` must be used to create an RDD. Then, convert it to a PySpark data frame using the spark session's `createDatFrame()` function.
+
+To achieve this, let's look at the steps below:
+#### 1. We will first import the libraries 
+Importing libraries is as shown in the code below:
+```Python
+from pyspark import SparkContext
+from pyspark.sql import SparkSession
+```
+#### 2. Create a spark context 
+This is as shown in the code below:
+```python
+sparkContext = SparkContext.getOrCreate()
+```
+To get a spark context for our experiment, we used `getOrCreate()` method. Depending on whether we've previously built one, we may use the `getOrCreate()` function to build or get an instance of a spark context.
+#### 3. Create a spark session
+Passing a string as an input to `appName()` will allow us to give our application a name. Spark session will be instantiated using `getOrCreate()`, which is the next step. `GetOrCreate()` will utilize an existing spark session if one is available, or it will build a new one if there is no spark session available.
+```python
+sparkSession = SparkSession.builder.appName('Create PySpark DataFrame From an Existing RDD ').getOrCreate()
+```
+#### 4. Create the Resilient Distributed Dataset
+```python
+resilientDistributedDataset = sparkContext.parallelize([('C',80,70,85,91), ('B',80,70,85,91), ("A", 80,78,96,92), ("A", 92,70,89,96)], 4)
+```
+The spark context's `parallelize()` method was utilized to parallelize the calculation.
+
+The `resilientDistributedDataset` data type will determine RDD creation success or failure. We should get `pyspark.resilientDistributedDataset.RDD` as a result. This is seen in the code below:
+```Python
+print(type(resilientDistributedDataset))
+```
+Output:
+```bash
+<class 'pyspark.rdd.RDD'>
+```
+#### 5. Convert the Resilient Distributed Dataset into PySpark DataFrame
+```python
+subjects = ['Division','Kiswahili','Biology','Agriculture','Mathematics']
+ourMarks_df = sparkSession.createDataFrame(resilientDistributedDataset, schema=subjects)
+```
+In this case, the RDD is sent in the form of data. The schema parameter of the `createDataFrame()` method will be filled with previously produced string topics.
+
+We'll look at the data type of `ourMarks_df` below to see if everything went well. 
+```Python
+print(type(ourMarks_df))
+```
+Output:
+```bash
+<class 'pyspark.sql.dataframe.DataFrame'>
+```
+Using the `printSchema()` method, we can analyze our schema, which is useful when we have many columns.

```

Comment:
Using the `printSchema()` method, we can analyze the schema, which is useful when we have many columns.